.zenodo.json

```diff
@@ -1,7 +1,7 @@
 {
   "creators": [
     {
-      "name": "The Turing Way Community"
+      "name": "The Turing Way Community, ."
```

Comment:
Hmmm, I don't believe this change should have been made

Reply:
Hmm strange. I'll take a look at this.

Reply:
No need! I opened this PR that should resolve it https://github.com/alan-turing-institute/the-turing-way/pull/2413

Reply:
@sgibson91 Ah that's great! I am curious about how you resolved it too. 

I can reproduce it locally by running `cffconvert` but it doesn't seem like it is caused by any of the changes in the PR. It looks like it is generating ", ." based on the `family-names` and `given-names` under `authors`. 

I noticed the original code in the main branch also has "." as the `given-names` but the zenodo.json is created correctly so I am thinking whether it is something to do with `cffconvert` but I don't see anything mentioned about this issue in the `cffconvert` Github project. It could be that it was deleted manually in the created zenodo.json before the json file is commited and pushed?

With "Jack" as the `given-names`:
<img width="1285" alt="Screenshot 2022-05-30 at 18 52 52" src="https://user-images.githubusercontent.com/9934371/171041546-46eed11c-d47f-43e6-8c05-5faf51d64e02.png">
<img width="1336" alt="Screenshot 2022-05-30 at 18 53 01" src="https://user-images.githubusercontent.com/9934371/171041566-9bf0b315-cc34-4ee6-a8c4-2a028d283f32.png">

With "." as the `given-names`:
<img width="755" alt="Screenshot 2022-05-30 at 19 01 10" src="https://user-images.githubusercontent.com/9934371/171041996-f74e4efe-ad5f-4db0-804e-06a436a0aff6.png">
<img width="1067" alt="Screenshot 2022-05-30 at 19 01 24" src="https://user-images.githubusercontent.com/9934371/171041998-8dbb3bac-042d-4042-b1c8-f617b9d91940.png">

Reply:
I fixed it by reading the file with a YAML reader instead of reading it line-by-line. The CITATION.cff file is a valid YAML file and can be treat as a Python dictionary when it's read in by a YAML parser :) No need to search for `.` or anything else

Reply:
Actually, I think it is cffconvert - my PR won't resolve the issue (but it is a much cleaner approach to editing the citation.cff file)

Reply:
Completely agree with your approach to using the YAML parser! I've looked at your PR and left some comments.

Our code doesn't search for `.` so it should due to the `cffconvert`. Can you try running `cffconvert --format zenodo > .zenodo.json` and see if the generated `.zenodo.json` has that issue too? 

If yes then we can have a script after the output of `.zenodo.json` to strip out the `, .` to fix it. A bit of a hacky way to do it though. Open to ideas if you know a cleaner way to do it!

Reply:
> Can you try running cffconvert --format zenodo > .zenodo.json and see if the generated .zenodo.json has that issue too?

Yes, the .zenodo.json file has this issue, so this is stemming from cffconvert. Does given names have to be present in the CITATION.cff file for it to be valid? We could just get rid of that field if not?

Reply:
It's good to know that removing "given_names" field fixes the issue! I did think about it but didn't try it in the end since I thought it wouldn't be valid after reading the documentation on the CFF format. It mentions that `family-names` and `given-names` are mandatory for a Person object.

![given_names](https://user-images.githubusercontent.com/9934371/171391087-95ba5fdd-9254-4060-afef-27241ee221a8.PNG)

Link to docs:
https://citation-file-format.github.io/assets/pdf/cff-specifications-1.0.3.pdf

I am surprised cffconvert --validate showed that it was! 

I'll mention that in https://github.com/alan-turing-institute/the-turing-way/pull/2424 too